tests/stepDefinitions/addPromotionContext.js

```diff
@@ -0,0 +1,28 @@
+const { Given, When, Then } = require('@cucumber/cucumber')
+const { expect } = require("@playwright/test");
+
+const { LoginPage} = require("../pageObjects/LoginPage");
+const { AddPromotion } = require("../pageObjects/AddPromotion");
+const addPromotion = new AddPromotion();
+const login = new LoginPage();
+
+
+Given('user {string} has logged in to the admin panel with following credentials:', async function (user, credentials) {
+    await login.navigateToAdminLoginPage();
+    for (const { email, password } of credentials.hashes()) {
+        await login.login(email,password);
+      }
+    await page.waitForURL("http://127.0.0.1:3000/", { timeout: 5000 });
+  });
+
+When('user {string} adds a new promotion with following details:', async function (user, details) {
+  await addPromotion.navigateToPromotionPage();
+  for (const { name, code, path, description, limit } of details.hashes()) {
```

Comment:
Use loop only when you are adding multiple items.
```suggestion
  for (const { name, code, path, description, limit } of details.hashes()) {
```